v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/transforms/ReadWithUniformPartitions.java

```diff
@@ -182,7 +197,9 @@ public PCollection<T> expand(PBegin input) {
     PCollection<Range> rangesToRead =
         peekRanges(mergedRanges).apply(ParDo.of(new UnflattenRangesDoFn()));
     return rangesToRead
-        .apply(getTransformName("ReshuffleFinal", null), Reshuffle.viaRandomKey())
+        .apply(
+            getTransformName("ReshuffleFinal", null),
+            Reshuffle.<Range>viaRandomKey().withNumBuckets(dbParallelizationForReads()))
```

Comment:
this number should be lower as we may have a bunch of readers on a given MySQL instance for optimal usage. But i suggest we take this optimization up as as a follow up.

Reply:
```
this number should be lower as we may have a bunch of readers on a given MySQL instanc
```
That's right, we need to make it more configurable and also need to get more concrete benchmarking

`suggest we take this optimization up as as a follow up.`
Sure!